test/e2e/common/secrets.go

```diff
@@ -313,3 +285,88 @@ func doSecretE2EWithMapping(f *framework.Framework, mode *int32) {
 
 	f.TestContainerOutput("consume secrets", pod, 0, expectedOutput)
 }
+
+//Existence of a second secret with same name and different namespace shouldn't affect the mount
+func doSecretE2EForMountNameSpaceIssue(f *framework.Framework, defaultMode *int32) {
+	var (
+		name            = "secret-test-" + string(uuid.NewUUID())
+		volumeName      = "secret-volume"
+		volumeMountPath = "/etc/secret-volume"
+		namespace1      = f.Namespace
+		namespace2      *api.Namespace
+		err             error
+	)
+
+	if namespace2, err = f.CreateNamespace("secret-namespace", nil); err != nil {
+		framework.Failf("unable to create new namespace %s: %v", namespace2.Name, err)
+	}
+
+	secret1, secret2 := secretForTest(namespace1.Name, name), secretForTest(namespace2.Name, name)
+
+	By(fmt.Sprintf("Creating secrets with name %s and namespaces %s %s", name, namespace1.Name, namespace2.Name))
+
+	if secret1, err = f.ClientSet.Core().Secrets(namespace1.Name).Create(secret1); err != nil {
+		framework.Failf("unable to create test secret %s: %v", secret1.Name, err)
+	}
+
+	if secret2, err = f.ClientSet.Core().Secrets(namespace2.Name).Create(secret2); err != nil {
+		framework.Failf("unable to create test secret %s: %v", secret2.Name, err)
+	}
+
+	secret2.Data = map[string][]byte{
+		"this_should_not_match_previous_info": []byte("similarly_this_should_not_match_previous_info\n"),
+	}
+
```

Comment:
Why not create just this secret and call `doSecretE2EWithoutMapping()`?

If you modify `doSecretE2EWithoutMapping()` to take the secret name as param (or nil and the same behavior is kept) and just call it will avoid the code duplication and test this just fine, right?

I think that is the most straight forward way to do it (just make sure the secrets have the same name). Am I missing something? What do you think about this?